### PR TITLE
Stochastic results

### PR DESCRIPTION
**Pull request by jyan1999**
# Description

Fixed the issue that when an agent if reloaded from a checkpoint, it produces deterministic actions. 

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/human_aware_rl/rllib/rllib.py [hunk 1]
```diff
@@ -93,15 +94,24 @@ def action(self, state):
...
             np.array([my_obs]), self.rnn_state
         )
-        agent_action = Action.INDEX_TO_ACTION[action_idx]
 
         # Softmax in numpy to convert logits to normalized probabilities
         logits = info["action_dist_inputs"]
         action_probabilities = softmax(logits)
 
+        # reseed to make sure the outputs are stochastic
+        random.seed()
```
Comment (micahcarroll) [inline src/human_aware_rl/rllib/rllib.py:107]: See convos on slack re: this line

Comment (jyan1999) [reply src/human_aware_rl/rllib/rllib.py]: Yeah sorry you are right. I was looking at the wrong results when I was testing it that night and got myself confused.

File: src/human_aware_rl/rllib/rllib.py [hunk 2]
```diff
@@ -93,15 +94,22 @@ def action(self, state):
...
-        agent_action = Action.INDEX_TO_ACTION[action_idx]
 
         # Softmax in numpy to convert logits to normalized probabilities
         logits = info["action_dist_inputs"]
         action_probabilities = softmax(logits)
 
+        # The original design is stochastic across different games,
+        # Though if we are reloading from a checkpoint it would inherit the seed at that point, producing deterministic results
+        [action_idx] = random.choices(
+            [0, 1, 2, 3, 4, 5], action_probabilities[0]
```
Comment (micahcarroll) [inline src/human_aware_rl/rllib/rllib.py:109]: Minor change here to make more robust to potential future changes
```suggestion
            list(range(Action.NUM_ACTIONS)), action_probabilities[0]
```

**Review by micahcarroll — APPROVED** [review]
LGTM! Thanks for the changes! You can go ahead and merge after you make the final suggested change (assuming it looks good to you)

**Review by micahcarroll — APPROVED** [review]
LGTM!

**Pull request merged**